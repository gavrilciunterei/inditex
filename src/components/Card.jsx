import React from 'react';
import { Link } from 'react-router-dom';

function Card({ img, id, model, brand, price }) {
  return (
    <Link to={`/detail/${id}`}>
      <div className="p-4">
        <div className="w-72 mx-auto overflow-hidden  rounded-lg shadow-lg bg-white transform hover:scale-105 transition duration-500">
          <div className="flex  justify-center">
            <img
              className="object-cover w-32 h-auto overflow-hidden"
              src={img}
              alt="avatar"
            />
          </div>

          <div className="flex flex-col pt-5 text-center">
            <span className="text-2xl font-bold text-black">{model}</span>
            <span className="text-sm text-black">{brand}</span>
            <span className=" bg-teal-100 text-teal-700 font-bold">
              {price.length > 0 ? price + 'â‚¬' : 'No disponible'}
            </span>
          </div>
        </div>
      </div>
    </Link>
  );
}

export default Card;
