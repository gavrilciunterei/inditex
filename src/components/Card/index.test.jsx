import React from 'react';
import renderer from 'react-test-renderer';
import Card from './index';
import { MemoryRouter } from 'react-router';

it('Test Card with price', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Card
          img="http://img.com"
          id="1"
          model="Predator"
          brand="Acer"
          price="100"
        />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Test Card without price', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Card
          img="http://img.com"
          id="1"
          model="Predator"
          brand="Acer"
          price=""
        />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
