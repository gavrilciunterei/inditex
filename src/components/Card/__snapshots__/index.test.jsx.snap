// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card with price 1`] = `
<a
  href="/detail/1"
  onClick={[Function]}
>
  <div
    className="p-4"
  >
    <div
      className="w-72 mx-auto overflow-hidden  rounded-lg shadow-lg bg-white transform hover:scale-105 transition duration-500"
    >
      <div
        className="flex  justify-center"
      >
        <img
          alt="avatar"
          className="object-cover w-32 h-auto overflow-hidden"
          src="http://img.com"
        />
      </div>
      <div
        className="flex flex-col pt-5 text-center"
      >
        <span
          className="text-2xl font-bold text-black"
        >
          Predator
        </span>
        <span
          className="text-sm text-black"
        >
          Acer
        </span>
        <span
          className=" bg-teal-100 text-teal-700 font-bold"
        >
          100â‚¬
        </span>
      </div>
    </div>
  </div>
</a>
`;

exports[`Test Card without price 1`] = `
<a
  href="/detail/1"
  onClick={[Function]}
>
  <div
    className="p-4"
  >
    <div
      className="w-72 mx-auto overflow-hidden  rounded-lg shadow-lg bg-white transform hover:scale-105 transition duration-500"
    >
      <div
        className="flex  justify-center"
      >
        <img
          alt="avatar"
          className="object-cover w-32 h-auto overflow-hidden"
          src="http://img.com"
        />
      </div>
      <div
        className="flex flex-col pt-5 text-center"
      >
        <span
          className="text-2xl font-bold text-black"
        >
          Predator
        </span>
        <span
          className="text-sm text-black"
        >
          Acer
        </span>
        <span
          className=" bg-teal-100 text-teal-700 font-bold"
        />
      </div>
    </div>
  </div>
</a>
`;
